lib/msal-browser/src/interaction_client/NativeInteractionClient.ts

```diff
@@ -319,17 +319,34 @@ export class NativeInteractionClient extends BaseInteractionClient {
         const scopeSet = new ScopeSet(scopes);
         scopeSet.appendScopes(OIDC_DEFAULT_SCOPES);
 
-        if (request.prompt) {
+        const prompt = () => {
+            // If request is silent, prompt is always none
+            switch (this.apiId) {
+                case ApiId.ssoSilent:
+                case ApiId.acquireTokenSilent_silentFlow:
+                    this.logger.trace("initializeNativeRequest: silent request sets prompt to none");
+                    return PromptValue.NONE;
+                default:
+                    break;
+            }
+
+            // If request is interactive, check if prompt is allowed to go directly to native broker
+            if (!request.prompt) {
```

Comment:
Not sure I follow this given the comment. Is this saying request.prompt being falsey means the request shuold go to the broker?

Reply:
Yes, rephrased to clarify